components/admin-panel/sections/NotificationsSettings/ActivitySwitch.js

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { useMutation } from '@apollo/client';
+import { FormattedMessage } from 'react-intl';
+
+import { ActivityClasses, ActivityTypes } from '../../../../lib/constants/activities';
+import { API_V2_CONTEXT, gqlV2 } from '../../../../lib/graphql/helpers';
+
+import StyledSwitch from '../../../StyledSwitch';
+import { TOAST_TYPE, useToasts } from '../../../ToastProvider';
+
+import { accountActivitySubscriptionsFragment } from './fragments';
+
+const refetchEmailNotificationQuery = gqlV2/* GraphQL */ `
+  query NotificationsSettingsRefetchQuery($id: String!) {
+    account(id: $id) {
+      id
+      ...AccountActivitySubscriptionsFields
+    }
+  }
+  ${accountActivitySubscriptionsFragment}
+`;
+
+const toggleEmailNotificationMutation = gqlV2/* GraphQL */ `
+  mutation ToggleEmailNotification($type: ActivityAndClassesType!, $account: AccountReferenceInput) {
+    toggleEmailNotification(type: $type, account: $account) {
```

Comment:
As mentioned in the API, `toggle*` mutations often results in UI glitches that are not there with `set*`

Reply:
Agreed. I think I moved too much of the business logic to the backend in this one. I'll update the resolver.